### PR TITLE
Misc session fixes

### PR DESCRIPTION
**Pull request by paweldomas**
Extract fixes from https://github.com/jitsi/lib-jitsi-meet/pull/460 discovered while working on P2P for react-native. These commits are not directly related to p2p+react-native and can be useful in other cases.

### CODE REVIEW

File: modules/RTC/DataChannels.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ DataChannels.prototype.onDataChannel = function(event) {
     const self = this;
 
     dataChannel.onopen = function() {
-        logger.info('Data channel opened by the Videobridge!', dataChannel);
+        logger.info('Data channel opened by the Videobridge!');
```
Comment (saghul) [inline modules/RTC/DataChannels.js]: ❤️ ❤️ ❤️

File: modules/xmpp/JingleSessionPC.js [hunk 2]
```diff
@@ -1050,45 +1050,51 @@ export default class JingleSessionPC extends JingleSession {
...
-            sessionTerminate.up().c('text').t(text);
+    terminate(success, failure, options) {
+        if (this.state === JingleSessionState.ENDED) {
+            return;
         }
 
-        // Calling tree() to print something useful
-        sessionTerminate = sessionTerminate.tree();
-        logger.info('Sending session-terminate', sessionTerminate);
+        if (!options || Boolean(options.sendSessionTerminate) !== false) {
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: `if (!options || Boolean(options.sendSessionTerminate))` ?

File: modules/xmpp/JingleSessionPC.js [hunk 3]
```diff
@@ -1050,45 +1050,51 @@ export default class JingleSessionPC extends JingleSession {
...
+                    .t(options.reasonDescription);
+            }
 
-        this.connection.sendIQ(
-            sessionTerminate,
-            success,
-            this.newJingleErrorHandler(sessionTerminate, failure), IQ_TIMEOUT);
+            // Calling tree() to print something useful
+            sessionTerminate = sessionTerminate.tree();
+            logger.info('Sending session-terminate', sessionTerminate);
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: info? maybe debug?

Comment (paweldomas) [inline modules/xmpp/JingleSessionPC.js]: session-initiate is logged on info, so I wanted to have it consistent. But I can change both eventually

Comment (paweldomas) [inline modules/xmpp/JingleSessionPC.js]: source-add and source-remove are also logged on info

**Comment by paweldomas** [comment]
@saghul I will do a force push for the comments as I think it makes sense to "rebase and merge" instead of "squash and merge" in this case. It will be good to have each of these fixes in separate commit.

**Comment by paweldomas** [comment]
@saghul will you have some time to take a look on the fixes and merge eventually ?

File: modules/xmpp/JingleSessionPC.js [hunk 4]
```diff
@@ -1310,9 +1356,19 @@ export default class JingleSessionPC extends JingleSession {
      *  rejects with an error {string}
      */
     _renegotiate(optionalRemoteSdp) {
+        const defaultRemoteSdp = this.peerconnection.remoteDescription.sdp;
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: Can we rename this to `currentRemoteSdp` instead? I think it reads a bit better

File: modules/xmpp/JingleSessionPC.js [hunk 5]
```diff
@@ -1310,9 +1356,19 @@ export default class JingleSessionPC extends JingleSession {
...
+
+        // FIXME new SDP() may not be necessary - raw SDP is needed anyway
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: can we check this and get rid of it if necessary since we're here?

File: modules/xmpp/JingleSessionPC.js [hunk 6]
```diff
@@ -1461,7 +1517,11 @@ export default class JingleSessionPC extends JingleSession {
                 this.peerconnection.addTrack(newTrack);
             }
 
-            if ((oldTrack || newTrack) && oldLocalSdp) {
+            // FIXME it makes sense to check for ACTIVE state in other places,
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: I don't really get what this fixme says, can you rephrase?

File: modules/xmpp/JingleSessionPC.js [hunk 7]
```diff
@@ -713,7 +713,8 @@ export default class JingleSessionPC extends JingleSession {
                         error,
                         this.mediaConstraints);
                     finishedCallback(error);
-                });
+                },
+                this.mediaConstraints);
```
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: This one looks like a "leftover bug" from "fix(JingleSessionPC): execute 'invite' on the queue" right? If so, can you please combine the commits?

File: modules/xmpp/JingleSessionPC.js (hunk 3 above)
Comment (saghul) [inline modules/xmpp/JingleSessionPC.js]: ok

File: modules/xmpp/JingleSessionPC.js (hunk 6 above)
Comment (paweldomas) [inline modules/xmpp/JingleSessionPC.js]: I'm removing it
